### PR TITLE
Parser: Fix - PAPP-23675 FIPS backward compatibility

### PR DESCRIPTION
**Pull request by paulnawat**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- md5 should be used for non-FIPs instances. sha256 should be used for FIPs instance.

## What is the current behavior? (OPTIONAL)
- sha256 is being used only to parse emails.

## What is the new behavior? (OPTIONAL)
- md5 should be used for non-FIPs instances. sha256 should be used for FIPs instance.

---
Thanks for contributing!

### CODE REVIEW

File: parser_email.py [hunk 1]
```diff
@@ -1347,8 +1347,21 @@ def _set_sdi(default_id, input_dict):
     return phantom.APP_SUCCESS
 
 
-def _create_dict_hash(input_dict):
+def _get_fips_enabled(self):
```
Comment (paulnawat) [inline parser_email.py:1350]: Helpers and logics are taken from https://github.com/splunk-soar-connectors/gsgmail/pull/2/files

The team agrees to repeat the helpers here, as we don't have shared modules as of now. See further discussion on Slack thread.

**Comment by paulnawat** [comment]
For your info, the Integration test failure here is due to a separate ticket which is not related to the FIPS compliance change here.

**Review by rgil-splunk — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

**Comment by gary-phantom** [comment]
playbook failed on integration tests.

**Review by alexa-phantom — APPROVED** [review]

**Comment by paulnawat** [comment]
@gary-phantom 

Thank you for having a look.
Parser playbook 003 has been recently updated 1 day ago. 
I've done sanity check by testing 003 against the latest `next` (i.e. the one where pipeline's green).
I was able to get the same 2 errors on both.

As a result, I reached out to @splunk-heena who made the 003 update to confirm if this is accurate on her side. 
Thanks @splunk-heena for confirming that this is indeed expected, as the app change required for the updated playbook has not been merged. 

As a result, we should be good to go with this FIPs change, since it's unrelated to the failure.

**Pull request merged**